react/OrderCoupon.tsx

```diff
@@ -1,54 +1,59 @@
-import React, { createContext, ReactNode, useContext, useState } from 'react'
+import React, {
+  createContext,
+  ReactNode,
+  useContext,
+  useState,
+  useCallback,
+  useRef,
+  useEffect,
+} from 'react'
 import { compose, graphql } from 'react-apollo'
 import { useOrderQueue } from 'vtex.order-manager/OrderQueue'
+import { useOrderForm } from 'vtex.order-manager/OrderForm'
 
 import InsertCoupon from './graphql/insertCoupon.graphql'
 
 interface Context {
-  insertCoupon: (coupon: string) => PromiseLike<void>
   coupon: string
-  isShowingPromoButton: boolean
+  showPromoButton: boolean
   errorKey: string
   changeCoupon: (coupon: string) => void
-  handleCouponChange: (evt: any) => void
   resetCouponInput: () => void
-  setIsShowingPromoButton: (value: boolean) => void
+  setShowPromoButton: (value: boolean) => void
   submitCoupon: (evt: any) => void
 }
 
 interface OrderItemsProviderProps {
   children: ReactNode
   InsertCoupon: any
+  CouponQuery: any
   coupon: string
   changeCoupon: (coupon: string) => void
 }
 
 const NO_ERROR = ''
-const CODE_DOESNT_EXIST = `CodeDoesntExist`
 
 const OrderCouponContext = createContext<Context | undefined>(undefined)
 
 export const OrderCouponProvider = compose(
   graphql(InsertCoupon, { name: 'InsertCoupon' })
 )(({ children, InsertCoupon }: OrderItemsProviderProps) => {
-  const { enqueue } = useOrderQueue()
-  const [coupon, setCoupon] = useState('')
-  const [isShowingPromoButton, setIsShowingPromoButton] = useState(true)
+  const { enqueue, listen } = useOrderQueue()
+  const { orderForm, setOrderForm } = useOrderForm()
+  const [coupon, setCoupon] = useState(
+    orderForm.marketingData.coupon ? orderForm.marketingData.coupon : ''
+  )
+  const [showPromoButton, setShowPromoButton] = useState(true)
   const [errorKey, setErrorKey] = useState(NO_ERROR)
 
   const changeCoupon = (coupon: string) => {
     setCoupon(coupon)
   }
 
-  const handleCouponChange = (evt: any) => {
-    evt.preventDefault()
-    const newCoupon = evt.target.value.trim()
-    changeCoupon(newCoupon)
-  }
-
   const resetCouponInput = () => {
     changeCoupon('')
-    setIsShowingPromoButton(false)
+    insertCoupon('')
```

Comment:
what is the difference between `changeCoupon('')` and `insertCoupon('')`?

Reply:
changeCoupon only sets the coupon info stored in OrderCoupon. InsertCoupon triggers the mutation.

Reply:
although it makes more sense to rename changeCoupon to setCoupon.